File: languages/go/entity/availability/ipns_publisher.go
package availability

import (
	"github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"
)

// IpnsPublisher represents a publisher for IPNS data availability.
type IpnsPublisher struct {
	Type proto.DataAvailabilityType
	Args PublisherArgs
}

// NewIpnsPublisher represents a publisher for IPNS data availability with creation option.
func NewIpnsPublisher() IpnsPublisher {
	return IpnsPublisher{
		Type: proto.DataAvailabilityType_IPNS,
		Args: PublisherArgs{
			IpnsKey: nil,
		},
	}
}

// UpdateIpnsPublisher represents a publisher for IPNS data availability with update option.
func UpdateIpnsPublisher(ipnsKey *IpnsKey) IpnsPublisher {
	return IpnsPublisher{
		Type: proto.DataAvailabilityType_IPNS,
		Args: PublisherArgs{
			IpnsKey: ipnsKey,
		},
	}
}

func (e IpnsPublisher) ToProto() *proto.Publisher {
	return &proto.Publisher{
		Type: e.Type,
		Args: e.Args.ToProto(),
	}
}

File: languages/js/src/entity/availability/ipns_publisher.ts
import * as proto from "../../bridge/proto/bloock_availability_entities";
import { IpnsKey } from "./ipns_key";
import { Publisher } from "./publisher";
import { PublisherArgs } from "./publisher_args";

/**
 * Represents a publisher for IPNS data availability.
 */
export class IpnsPublisher implements Publisher {
  type: proto.DataAvailabilityType;
  args: PublisherArgs;

  /**
   * Constructs a IpnsPublisher object with the specified parameters.
   */
  constructor(ipnsKey: IpnsKey) {
    this.type = proto.DataAvailabilityType.IPNS;
    this.args = new PublisherArgs(ipnsKey);
  }

  public toProto(): proto.Publisher {
    return proto.Publisher.fromPartial({
      type: this.type,
      args: this.args.toProto()
    });
  }
}